### DISCUSSION TITLE
Setting up a Blacklist

### DISCUSSION THREAD

**Discussion by ESNFrankfurt** [pretix/pretix#2623] [Ideas]
Hey Everyone!
so my association has been looking for a solution since there's some people who were deemed unfit to join our events due to various reasons. Now we would like to implement a Blacklist using these people's phone numbers, allowing pretix to automatically reject the attempt of signing up to an event from a person who registers with said number. Preferrably this would work in combination with an email blacklist but lets face it an email blacklist would be super easy to circumvent.

Anyone able to help us out with that? 

Cheers
Leo #

**Comment by raphaelm** [comment]
Yes, email blacklists would be trivial to circumvent, which is why we don't have them.

Phone number could be a little more effective **but** only if it is actually verified (i.e. by sending an SMS verification code), which is actually not implemented. If I can just enter any number, it is just as ineffective.

I don't think this is something we'll ever have in pretix, but something that can be easily added as a plugin.

**Comment by listerr** [comment]
I looked at the email restrictions plugin for something similar.

It only allows a whitelist, not a blacklist. It would be good if it had some blacklist, but just submitted the order for manual approval.
(And maybe an option to submit for manual approval anyone who is not on the whitelist.)

I think you could sort-of do something with it though.

In Email Restrictions Plugin, set maximum orders per email address to 1

Create a product called "Banned Users" than doesn't show up in the shop, (or only pretixPOS)
 - Unset "Allow product to be canceled or changed"
 - Tickets and Checkin -> Generate tickets = Never   /  [X] Requires special attention.

Create a quota if needed: Banned, unlimited. [X] Ignore this quota when determining event availability.
 - Add the "Banned Users" product to the quota.

- Import your list of banned users as orders for the product "Banned Users"

(Or if there's only a few, create some orders with offline order plugin and then amend the details with the email address)

Now if they try to order from a banned email address, it will not accept the order because they already have an "order".

If they turn up anyway and someone puts their name into the Check-in, it will show as Banned user / requires special attention.

Of course, this is a hack and if you actually do want to allow multiple orders / multiple products from the same email address, won't work. 

May be better off to write something that works with the API, checks any orders that are from banned people, and cancels the order or flags it for attention.
